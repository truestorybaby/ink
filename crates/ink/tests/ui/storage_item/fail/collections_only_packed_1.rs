use ink_prelude::vec::Vec;
use ink_storage::Lazy;

#[ink::storage_item]
struct NonPacked {
    a: Lazy<u128>,
}

#[ink::storage_item]
struct Contract {
    a: Vec<NonPacked>,
}

fn main() {}
