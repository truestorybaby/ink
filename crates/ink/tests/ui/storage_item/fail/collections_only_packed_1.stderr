error[E0277]: the trait bound `Vec<NonPacked>: parity_scale_codec::Decode` is not satisfied
  --> tests/ui/storage_item/fail/collections_only_packed_1.rs:11:8
   |
11 |     a: Vec<NonPacked>,
   |        ^^^ the trait `parity_scale_codec::Decode` is not implemented for `Vec<NonPacked>`
   |
   = help: the trait `parity_scale_codec::Decode` is implemented for `Vec<T>`
   = note: required for `Vec<NonPacked>` to implement `Packed`
   = note: required for `Vec<NonPacked>` to implement `StorableHint<()>`
   = note: required for `Vec<NonPacked>` to implement `AutoStorableHint<ManualKey<453539032>>`

error[E0277]: the trait bound `[NonPacked]: Encode` is not satisfied
  --> tests/ui/storage_item/fail/collections_only_packed_1.rs:11:8
   |
11 |     a: Vec<NonPacked>,
   |        ^^^ the trait `Encode` is not implemented for `[NonPacked]`
   |
   = help: the following other types implement trait `Encode`:
             [T; N]
             [T]
   = note: required for `Vec<NonPacked>` to implement `Encode`
   = note: required for `Vec<NonPacked>` to implement `Packed`
   = note: required for `Vec<NonPacked>` to implement `StorableHint<()>`
   = note: required for `Vec<NonPacked>` to implement `AutoStorableHint<ManualKey<453539032>>`

error[E0277]: the trait bound `Vec<NonPacked>: parity_scale_codec::Decode` is not satisfied
  --> tests/ui/storage_item/fail/collections_only_packed_1.rs:10:8
   |
10 | struct Contract {
   |        ^^^^^^^^ the trait `parity_scale_codec::Decode` is not implemented for `Vec<NonPacked>`
   |
   = help: the trait `parity_scale_codec::Decode` is implemented for `Vec<T>`
   = note: required for `Vec<NonPacked>` to implement `Packed`
   = note: required for `Vec<NonPacked>` to implement `StorableHint<()>`
   = note: required for `Vec<NonPacked>` to implement `AutoStorableHint<ManualKey<453539032>>`
note: required because it appears within the type `Contract`
  --> tests/ui/storage_item/fail/collections_only_packed_1.rs:10:8
   |
10 | struct Contract {
   |        ^^^^^^^^
note: required by a bound in `Storable`
  --> $WORKSPACE/crates/storage/traits/src/storage.rs
   |
   | pub trait Storable: Sized {
   |                     ^^^^^ required by this bound in `Storable`

error[E0277]: the trait bound `[NonPacked]: Encode` is not satisfied
  --> tests/ui/storage_item/fail/collections_only_packed_1.rs:10:8
   |
10 | struct Contract {
   |        ^^^^^^^^ the trait `Encode` is not implemented for `[NonPacked]`
   |
   = help: the following other types implement trait `Encode`:
             [T; N]
             [T]
   = note: required for `Vec<NonPacked>` to implement `Encode`
   = note: required for `Vec<NonPacked>` to implement `Packed`
   = note: required for `Vec<NonPacked>` to implement `StorableHint<()>`
   = note: required for `Vec<NonPacked>` to implement `AutoStorableHint<ManualKey<453539032>>`
note: required because it appears within the type `Contract`
  --> tests/ui/storage_item/fail/collections_only_packed_1.rs:10:8
   |
10 | struct Contract {
   |        ^^^^^^^^
note: required by a bound in `Storable`
  --> $WORKSPACE/crates/storage/traits/src/storage.rs
   |
   | pub trait Storable: Sized {
   |                     ^^^^^ required by this bound in `Storable`

error[E0277]: the trait bound `Vec<NonPacked>: parity_scale_codec::Decode` is not satisfied
  --> tests/ui/storage_item/fail/collections_only_packed_1.rs:9:1
   |
9  | #[ink::storage_item]
   | ^^^^^^^^^^^^^^^^^^^^ the trait `parity_scale_codec::Decode` is not implemented for `Vec<NonPacked>`
   |
   = help: the trait `parity_scale_codec::Decode` is implemented for `Vec<T>`
   = note: required for `Vec<NonPacked>` to implement `Packed`
   = note: required for `Vec<NonPacked>` to implement `StorableHint<()>`
   = note: required for `Vec<NonPacked>` to implement `AutoStorableHint<ManualKey<453539032>>`
note: required because it appears within the type `Contract`
  --> tests/ui/storage_item/fail/collections_only_packed_1.rs:10:8
   |
10 | struct Contract {
   |        ^^^^^^^^
note: required by a bound in `Result`
  --> $RUST/core/src/result.rs
   |
   | pub enum Result<T, E> {
   |                 ^ required by this bound in `Result`
   = note: this error originates in the derive macro `::ink::storage::traits::Storable` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `[NonPacked]: Encode` is not satisfied
  --> tests/ui/storage_item/fail/collections_only_packed_1.rs:9:1
   |
9  | #[ink::storage_item]
   | ^^^^^^^^^^^^^^^^^^^^ the trait `Encode` is not implemented for `[NonPacked]`
   |
   = help: the following other types implement trait `Encode`:
             [T; N]
             [T]
   = note: required for `Vec<NonPacked>` to implement `Encode`
   = note: required for `Vec<NonPacked>` to implement `Packed`
   = note: required for `Vec<NonPacked>` to implement `StorableHint<()>`
   = note: required for `Vec<NonPacked>` to implement `AutoStorableHint<ManualKey<453539032>>`
note: required because it appears within the type `Contract`
  --> tests/ui/storage_item/fail/collections_only_packed_1.rs:10:8
   |
10 | struct Contract {
   |        ^^^^^^^^
note: required by a bound in `Result`
  --> $RUST/core/src/result.rs
   |
   | pub enum Result<T, E> {
   |                 ^ required by this bound in `Result`
   = note: this error originates in the derive macro `::ink::storage::traits::Storable` (in Nightly builds, run with -Z macro-backtrace for more info)
